qiskit/circuit/controlflow/if_else.py

```diff
@@ -440,26 +438,22 @@ def __enter__(self):
             # I'm not even sure how you'd get this to trigger, but just in case...
             raise CircuitError("Cannot attach an 'else' to a broadcasted 'if' block.")
         appended = appended_instructions[0]
-        operation, _, _ = circuit._peek_previous_instruction_in_scope()
-        if appended is not operation:
+        instruction = circuit._peek_previous_instruction_in_scope()
+        if appended is not instruction:
```

Comment:
So we look at the references for the `CircuitInstruction` instances now, rather than the operation itself. That makes sense I think, but just wanted to call it out.

Reply:
Yeah, it probably always should have been this way - I'm fairly sure I should have been using identity of the tuple in the original code.